solana/bridge/program/tests/integration.rs

```diff
@@ -322,6 +323,84 @@ fn test_bridge_messages(context: &mut Context) {
     }
 }
 
+fn test_bridge_messages_unreliable(context: &mut Context) {
```

Comment:
Can we also add a test to ensure that messages with different sizes return an error?  And also one for messages with different emitters.  I think it's trivially obvious from reading the source that these will fail but the tests are mostly so that those checks don't get accidentally removed in the future.